debezium-server/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java

```diff
@@ -0,0 +1,126 @@
+import static java.net.HttpURLConnection.HTTP_OK;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.inject.Named;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.server.BaseChangeConsumer;
+
+/**
+ * Implementation of the consumer that delivers the messages to an HTTP Webhook destination.
+ *
+ * @author Chris Baumbauer
+ */
+@Named("http")
+@Dependent
+public class HttpChangeConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpChangeConsumer.class);
+
+    private static final String PROP_PREFIX = "debezium.sink.http.";
+    private static final String PROP_WEBHOOK_URL = "url";
+    private static final String PROP_CLIENT_TIMEOUT = "timeout.ms";
+
+    private static final Long HTTP_TIMEOUT = Integer.toUnsignedLong(60000); // Default to 60s
+
+    private static String sinkUrl;
+    private static Duration timeoutDuration;
+
+    // If this is running as a Knative object, then expect the sink URL to be located in `K_SINK`
+    // as per https://knative.dev/development/eventing/custom-event-source/sinkbinding/
+    @PostConstruct
+    void connect() {
+        final Config config = ConfigProvider.getConfig();
+        String sink = System.getenv("K_SINK");
+        timeoutDuration = Duration.ofMillis(HTTP_TIMEOUT);
+
+        if (sink != null) {
+            sinkUrl = sink;
+        }
+        else {
+            sinkUrl = config.getValue(PROP_PREFIX + PROP_WEBHOOK_URL, String.class);
+        }
+
+        config.getOptionalValue(PROP_PREFIX + PROP_CLIENT_TIMEOUT, String.class)
+                .ifPresent(t -> timeoutDuration = Duration.ofMillis(Long.parseLong(t)));
+
+        LOGGER.info("Using sink URL: {}", sinkUrl);
+    }
+
+    private String getContentType(boolean cloudevent) {
+        if (cloudevent) {
+            return "application/cloudevents+json";
+        }
+        else {
+            return "application/json";
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        HttpClient client = HttpClient.newHttpClient();
+
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+
+            try {
+                HttpRequest.Builder requestBuilder = HttpRequest.newBuilder(new URI(sinkUrl))
```

Comment:
So do we need a new request builder each time indeed? This seems a bit excessive?

Reply:
In this case more for setting the `Content-Header`, but can move it out.